### PR TITLE
Remove unreachable NSG code

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: PR #4085 added an if statement to remove NSG for AKS but that code is effectively unreachable because it is inside an "if hosted master statement"

Non-hosted master clusters have their own NSG definition in master resources:
https://github.com/Azure/acs-engine/blob/fix-nsg/parts/k8s/kubernetesmasterresources.t#L87

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: parts/k8s/kubernetesbase.t [hunk 1]
```diff
@@ -146,54 +146,7 @@
       "location": "[variables('location')]",
       "name": "[variables('nsgName')]",
       "properties": {
-        "securityRules": [
-{{if .HasWindows}}
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesbase.t]: AKS doesn't have Windows, so this never applies

File: parts/k8s/kubernetesbase.t [hunk 2]
```diff
@@ -146,54 +146,7 @@
...
-              "destinationPortRange": "3389-3389",
-              "direction": "Inbound",
-              "priority": 102,
-              "protocol": "Tcp",
-              "sourceAddressPrefix": "*",
-              "sourcePortRange": "*"
-            }
-          },
-{{end}}
-{{if not IsHostedMaster}}
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesbase.t]: this whole thing is inside a `{{if IsHostedMaster}}` block https://github.com/Azure/acs-engine/pull/4164/files#diff-66965012f8a909902930dd7fa6804d3eR96 so this is dead code

**Comment by CecileRobertMichon** [comment]
@weinong FYI

File: parts/k8s/kubernetesbase.t (hunk 1 above)
Comment (jackfrancis) [reply parts/k8s/kubernetesbase.t]: Can we indent all the conditional blocks so that it is easier to follow?

Comment (CecileRobertMichon) [reply parts/k8s/kubernetesbase.t]: Done. https://github.com/Azure/acs-engine/pull/4164/files?w=1 to see the diff without whitespaces.

**Pull request merged**